src/constants/themes/types.ts

```diff
@@ -16,6 +16,7 @@ type Typography =
   | typeof secondaryTheme['TYPOGRAPHY'];
 
 export type ThemeColors = valueof<Colors>;
+export type ThemeColorOrEmptyString = ThemeColors | '';
```

Comment:
Before this PR there were some colors with empty string value in the themes, now that we have the final structure with full values, it started to throw errors in buttons components where `textColor` uses empty string as  default value

Reply:
It looks like we can safely remove those default arguments and handle them as undefined. I don't think we should be handling empty strings--the types (and usage) should be strict.

Reply:
I initially did that, I found one conditional where it checks for `textColor === ''` 
I didnt wanted to touch the component logic, but I agree with you it will be cleaner to handle as undefined. Ill do this tommorrow when I am fresh.

Reply:
refactored this to use undefined